spec/java/lang/System.StdOut.lsl

```diff
@@ -0,0 +1,348 @@
+libsl "1.1.0";
+
+library std
+    version "11"
+    language "Java"
+    url "https://github.com/openjdk/jdk11/blob/master/src/java.base/share/classes/java/io/PrintStream.java";
+
+// imports
+
+import java/lang/CharSequence;
+import java/lang/Object;
+import java/lang/String;
+import java/util/Locale;
+
+import java/lang/System;
+import runtime/utils/VoidOutputStream;
+
+
+// automata
+
+automaton System_PrintStreamAutomaton
+(
+)
+: System_PrintStream
+{
+    // states and shifts
+
+    initstate Initialized;
+
+    shift Initialized -> self by [
+        // instance methods
+        append (System_PrintStream, CharSequence),
+        append (System_PrintStream, CharSequence, int, int),
+        append (System_PrintStream, char),
+        checkError,
+        close,
+        flush,
+        format (System_PrintStream, Locale, String, array<Object>),
+        format (System_PrintStream, String, array<Object>),
+        print (System_PrintStream, Object),
+        print (System_PrintStream, String),
+        print (System_PrintStream, boolean),
+        print (System_PrintStream, char),
+        print (System_PrintStream, array<char>),
+        print (System_PrintStream, double),
+        print (System_PrintStream, float),
+        print (System_PrintStream, int),
+        print (System_PrintStream, long),
+        printf (System_PrintStream, Locale, String, array<Object>),
+        printf (System_PrintStream, String, array<Object>),
+        println (System_PrintStream),
+        println (System_PrintStream, Object),
+        println (System_PrintStream, String),
+        println (System_PrintStream, boolean),
+        println (System_PrintStream, char),
+        println (System_PrintStream, array<char>),
+        println (System_PrintStream, double),
+        println (System_PrintStream, float),
+        println (System_PrintStream, int),
+        println (System_PrintStream, long),
+        write (System_PrintStream, array<byte>),
+        write (System_PrintStream, array<byte>, int, int),
+        write (System_PrintStream, int),
+    ];
+
+    // internal variables
+
+    var closed: boolean = false;
+    var error: boolean = false;
+
+
+    // utilities
+
+    @AutoInline @Phantom proc _throwNPE (): void
+    {
+        action THROW_NEW("java.lang.NullPointerException", []);
+    }
+
+
+    @AutoInline @Phantom proc _checkOpen (): void
+    {
+        if (this.closed)
+            this.error = true;
+    }
+
+
+    // constructors
+
+    // static methods
+
+    // methods
+
+    fun *.append (@target self: System_PrintStream, csq: CharSequence): PrintStream
+    {
+        if (csq == null)
+            _throwNPE();
+
+        _checkOpen();
+
+        result = self as Object as PrintStream;
```

Comment:
I'm not sure on how the current implementation of method dispatching works in USVM.
As I can see from `org.gradle.internal.io.LinePerThreadBufferingOutputStream` class (as the only `@Override`ing example I have) its implementation for method `append(java.lang.CharSequence)` return a reference to an object that is a sub-class of `java.io.PrintStream`, when the translated (using the LibSL translator tool) *implementation* class (i.e. `generated.java.lang.System_PrintStream`) is NOT.
The latter is important since we do not want to trigger `init` of `java.io.OutputStreamWriter` because of [Charset issues](https://github.com/UnitTestBot/java-stdlib-approximations/issues/3).